### PR TITLE
Adds App Bundle generation to the CLI

### LINKED ISSUE #119: Allow `build ` to optionally output an App Bundle
The Play Store warns developers to use App Bundles instead of APKs. 

The APK generated by llama-pack is already small and App Bundles are unlikely to generate gains. APKs are also easier to debug locally.

But we could provide developers with the choice of outputting a an App Bundle, by wrapping the proper gradle commands and providing a flab in the `build` command.

```shell
.\gradlew.bat bundleRelease
jarsigner -verbose -sigalg SHA256withRSA -digestalg SHA-256 -keystore android.keystore .\app\build\outputs\bundle\release\app-release.aab android
```

More Info:

- https://developer.android.com/studio/build/building-cmdline#build_bundle 
- https://docs.oracle.com/javase/8/docs/technotes/tools/windows/jarsigner.html 
- https://stackoverflow.com/questions/50560045/sign-android-app-bundle-from-command-line

### PR DESCRIPTION
**Pull request by andreban**
- Refactors `build.ts` to have a `Build` class
- Moves APK generation to a method
- Adds method to generate App Bundles
- Adds a `build` flag to generate App Bundles
- Fixes stacktraces for the failed validation showing in the wrong place.

Closes #119

### CODE REVIEW

File: packages/cli/src/lib/cmds/build.ts [hunk 1]
```diff
@@ -30,137 +30,170 @@ interface SigningKeyPasswords {
...
+  async buildApk(signingKey: SigningKeyInfo, passwords: SigningKeyPasswords): Promise<void> {
+    await this.gradleWrapper.assembleRelease();
+    await this.androidSdkTools.zipalign(
+        './app/build/outputs/apk/release/app-release-unsigned.apk', // input file
+        './app-release-unsigned-aligned.apk', // output file
+    );
+    const outputFile = './app-release-signed.apk';
+    await this.androidSdkTools.apksigner(
+        signingKey.path,
+        passwords.keystorePassword, // keystore password
```
Comment (PEConn) [inline packages/cli/src/lib/cmds/build.ts:133]: The these comments needed? Your variables are named well enough to be self explanatory.

File: packages/cli/src/lib/cmds/build.ts [hunk 2]
```diff
@@ -30,137 +30,170 @@ interface SigningKeyPasswords {
...
-      printValidationResult(pwaValidationResult, log);
-      if (pwaValidationResult.status === 'FAIL') {
-        log.warn('PWA Quality Criteria check failed.');
+    if (this.args.generateAppBundle) {
+      await this.buildAppBundle(twaManifest.signingKey, passwords);
+    }
+
+    await this.generateAssetLinks(twaManifest, passwords);
+
+    if (validationPromise) {
```
Comment (PEConn) [inline packages/cli/src/lib/cmds/build.ts:175]: Could you make this line a bit clearer - are we checking that the validationPromise is not null, or are we checking that it is not false?

File: packages/cli/src/lib/cmds/build.ts (hunk 1 above)
Comment (andreban) [reply packages/cli/src/lib/cmds/build.ts]: done.

File: packages/cli/src/lib/cmds/build.ts (hunk 2 above)
Comment (andreban) [reply packages/cli/src/lib/cmds/build.ts]: done - Changed to declare `let validationPromise = null` earlier so we can check `validationPromise !== null` here.

**Pull request merged**